x/ibc/05-port/keeper/keeper.go

```diff
@@ -33,7 +34,7 @@ func (k Keeper) isBounded(portID string) bool {
 // Ports must be bound statically when the chain starts in `app.go`.
 // The capability must then be passed to a module which will need to pass
 // it as an extra parameter when calling functions on the IBC module.
-func (k *Keeper) BindPort(portID string) sdk.CapabilityKey {
+func (k *Keeper) BindPort(ctx sdk.Context, portID string) capability.Capability {
```

Comment:
Wrote this function gleaning as much as i could from ICS-5 and capability implementation

From my understanding, a module will call this function. IBC will then create a capability and return it to the caller.

The calling module can then call `ClaimCapability` to own the capability and thus the port associated with the capability.

At this point, *both* the IBC module and the calling module own the capability and can authenticate it.

Is this the correct behavior?

Reply:
See [updated spec](https://github.com/cosmos/ics/tree/cwgoes/capability-rework/spec/ics-005-port-allocation#binding-to-a-port):

```typescript
function bindPort(id: Identifier): CapabilityKey {
    abortTransactionUnless(validatePortIdentifier(id))
    abortTransactionUnless(privateStore.get(portPath(id)) === null)
    capability = newCapability(portPath(id))
    return capability
}
```